src/Ng.TraceListeners/AzureTableStorageStatusTraceListener.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.WindowsAzure.Diagnostics;
+using Microsoft.WindowsAzure.ServiceRuntime;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Auth;
+using Microsoft.WindowsAzure.Storage.Table;
+using Ng.TraceListeners.Models;
+
+namespace Ng.TraceListeners
+{
+    /// <summary>
+    /// Saves application status data to Azure table storage.
+    /// Only messages in the following form are written to table storage.
+    ///   #Status #Activity=Download packages #Status=Begin #Details=Download source http://foo/downloads
+    /// Register the trace listener using the following. Specify the Azure table storage connection
+    /// string using the tablestorage field. Note that this trace listener uses * as the field delimiter
+    /// because the = character is used in the connection string.
+    /// <add name="AzureTableStatus" type="Ng.TraceListeners.AzureTableStorageStatusTraceListener, Ng.TraceListeners" initializeData="tablestorage*connectionstring" />
+    /// </summary>
+    public class AzureTableStorageStatusTraceListener : TraceListenerBase
+    {
+        CloudStorageAccount _storageAccount = null;
+        CloudTableClient _tableClient = null;
+        CloudTable _table = null;
+
+        public AzureTableStorageStatusTraceListener(String initializeData) : base(initializeData)
+        {
+            string connectionString = null;
+            string tableName = "status";
```

Comment:
Yes, "status" is just the default table name. You can override this in the config file of the app. The syntax would look something like this. Also I added this example to as a comment in the source file.
        <add name="AzureTableStatus" type="Ng.TraceListeners.AzureTableStorageStatusTraceListener, Ng.TraceListeners" initializeData="table*nugetindexerstatus,tablestorage*VeryLogConnectionString/" />